### PR TITLE
Add ads

### PR DESCRIPTION
**Pull request by gbaranski**

### CODE REVIEW

**Comment by gbaranski** [comment]
@stasbar please add `mixer_ad_provider` to remote config

**Comment by gbaranski** [comment]
I mean on the production version.

File: src/utils/firebase.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ export const batteriesStorageRef = prodStorage.ref('batteries').child('images');
 export const usersStorageRef = storage().ref('users').child('images');
 export const coilsStorageRef = storage().ref('coils').child('images');
 export const photosStorageRef = storage().ref('gears').child('images');
+export const bannersStorageRef = storage().ref('banners');
```
Comment (stasbar) [inline src/utils/firebase.ts:90]: unused, remove.

File: src/services/storage.ts [hunk 2]
```diff
@@ -60,10 +69,34 @@ function getDownloadUrl(
...
 }
+
+export interface BannerProperties {
+  name: string;
+  linkUrl: string;
+  imageGs: string;
+}
+
+export async function getAdImageProperties(parameterKey: string): Promise<BannerProperties> {
+  return JSON.parse(remoteConfig().getValue(parameterKey).asString()) as BannerProperties;
```
Comment (stasbar) [inline src/services/storage.ts:101]: I would split to two lines for better readability, but w/e.

File: src/app.tsx [hunk 3]
```diff
@@ -43,6 +44,7 @@ export async function getInitialState(): Promise<{
       if (!user) {
         // User is first time logged in
         user = await initializeUser(firebaseUser);
+        remoteConfig().fetchAndActivate();
```
Comment (stasbar) [inline src/app.tsx:47]: not sure if it's good place since this will be called each time user changes the page.

**Comment by stasbar** [comment]
You should also add default values for each key (in `src/utils/firebase.ts `)
```js
remoteConfig.defaultConfig = ({
  'batteries_database_ad_provider': '{    "name" : "admob", 	 "linkUrl" : "",    "imageGs" : "" }',
  'lobby_ad_provider': '{    "name" : "admob", 	 "linkUrl" : "",    "imageGs" : "" }',
  ...
});
```

**Comment by stasbar** [comment]
@gbaranski have you seen my comments ?

**Comment by gbaranski** [comment]
@stasbar what do you think it should be in defaults?

File: src/components/Banner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <>
+      {bannerProperties && (
+        <a href={bannerProperties.linkUrl}>
+          <img src={bannerSrc} alt={bannerProperties.name} />
+        </a>
+      )}
+    </>
+  );
```
[the remark this answers is not in the document]
Comment (gbaranski) [reply src/components/Banner/index.tsx:34]: @stasbar do u think that will be better in any way?

File: src/pages/batteries/Batteries.tsx [hunk 5]
```diff
@@ -16,6 +17,9 @@ const Batteries = () => {
   return (
     <PageContainer>
       <Row justify="center" gutter={32}>
+        <div style={{ marginBottom: '2%' }}>
+          <Banner providerName="batteries_database_ad_provider" />
+        </div>
```
[the remark this answers is not in the document]
Comment (gbaranski) [reply src/pages/batteries/Batteries.tsx:22]: @stasbar sometimes thereis marginTop

File: src/app.tsx (hunk 3 above)
Comment (gbaranski) [reply src/app.tsx:47]: but thats the same you do with retreiving users from database, basiccly initialstate executes rarely.

**Comment by stasbar** [comment]
`'{    "name" : "admob", 	 "linkUrl" : "",    "imageGs" : "" }'` for everyone. It defaults to admob if no provider is currently avaiable.

File: src/components/Banner/index.tsx [hunk 6]
```diff
@@ -14,13 +14,22 @@ export default (props: { providerName: string }) => {
   }
 
   useEffect(() => {
-    getAdImageProperties(props.providerName).then((adProperties) => {
-      console.log({ adProperties });
-      getBannerUrl(adProperties).then((newBannerSrc) => {
-        setBannerSrc(newBannerSrc);
-        setBannerProperties(adProperties);
+    getAdImageProperties(props.providerName)
```
Comment (stasbar) [inline src/components/Banner/index.tsx:17]: Does this method actually throw Exception ?

**Review by stasbar — COMMENTED** [review]
question

File: src/components/Banner/index.tsx (hunk 6 above)
Comment (gbaranski) [reply src/components/Banner/index.tsx:17]: yes @stasbar

**Pull request merged**